core/src/bls_sigverify/bls_sigverify_service.rs

```diff
@@ -0,0 +1,168 @@
+use {
+    crate::bls_sigverify::{bls_sigverifier::BLSSigVerifier, stats::BLSPacketStats},
+    agave_votor_messages::consensus_message::ConsensusMessage,
+    core::time::Duration,
+    crossbeam_channel::{Receiver, RecvTimeoutError, SendError, TrySendError},
+    solana_measure::measure::Measure,
+    solana_perf::{
+        deduper::{self, Deduper},
+        packet::PacketBatch,
+    },
+    solana_streamer::streamer::{self, StreamerError},
+    solana_time_utils as timing,
+    std::{
+        thread::{self, Builder, JoinHandle},
+        time::Instant,
+    },
+    thiserror::Error,
+};
+
+#[derive(Error, Debug)]
+pub enum BLSSigVerifyServiceError<SendType> {
+    #[error("send packets batch error")]
+    Send(Box<SendError<SendType>>),
```

Comment:
can we not use `#[from]` here and get rid of the `Box`? Why do we need the `Box`?

Reply:
Sure, actually we don't need SendError any more, these are all bounded channels so we need TrySendError. I moved the `error!` log to the caller.